examples/platforms/nrf528xx/nrf52840/nrf52840.cmake

```diff
@@ -0,0 +1,281 @@
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+set(OT_PLATFORM_LIB "openthread-nrf52840" "openthread-nrf52840-transport" PARENT_SCOPE)
+
+if(NOT OT_CONFIG)
+    set(OT_CONFIG "${PROJECT_SOURCE_DIR}/examples/platforms/nrf528xx/nrf52840/openthread-core-nrf52840-config.h")
+    set(OT_CONFIG ${OT_CONFIG} PARENT_SCOPE)
+endif()
+
+set(OT_BOOTLOADER "USB" CACHE STRING "OT bootloader type")
+if(${OT_BOOTLOADER} STREQUAL "USB")
+    list(APPEND OT_PLATFORM_DEFINES "APP_USBD_NRF_DFU_TRIGGER_ENABLED=1")
+    set(LD_FILE "${PROJECT_SOURCE_DIR}/examples/platforms/nrf528xx/nrf52840/nrf52840_bootloader_usb.ld")
+elseif(${OT_BOOTLOADER} STREQUAL "UART")
+    set(LD_FILE "${PROJECT_SOURCE_DIR}/examples/platforms/nrf528xx/nrf52840/nrf52840_bootloader_uart.ld")
+elseif(${OT_BOOTLOADER} STREQUAL "BLE")
+    set(LD_FILE "${PROJECT_SOURCE_DIR}/examples/platforms/nrf528xx/nrf52840/nrf52840_bootloader_ble.ld")
+else()
+    set(LD_FILE "${PROJECT_SOURCE_DIR}/examples/platforms/nrf528xx/nrf52840/nrf52840.ld")
+endif()
+
+list(APPEND OT_PLATFORM_DEFINES
+    "OPENTHREAD_CORE_CONFIG_PLATFORM_CHECK_FILE=\"openthread-core-nrf52840-config-check.h\""
+    "OPENTHREAD_SETTINGS_RAM=0"
+)
+if(OT_BUILTIN_MBEDTLS)
+    list(APPEND OT_PLATFORM_DEFINES "MBEDTLS_CONFIG_FILE=\"mbedtls-config.h\"")
+    list(APPEND OT_PUBLIC_INCLUDES
+        "${PROJECT_SOURCE_DIR}/third_party/mbedtls"
+        "${PROJECT_SOURCE_DIR}/third_party/mbedtls/repo/include"
+    )
+    if(${OT_MBEDTLS_THREADING})
+        list(APPEND OT_PLATFORM_DEFINES "MBEDTLS_THREADING_C")
+        list(APPEND OT_PLATFORM_DEFINES "MBEDTLS_THREADING_ALT")
+        list(APPEND OT_PUBLIC_INCLUDES
+            "${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/include/software-only-threading"
+        )
+    endif()
+else()
+    list(APPEND OT_PLATFORM_DEFINES "MBEDTLS_CONFIG_FILE=\"nrf-config.h\"")
+    list(APPEND OT_PUBLIC_INCLUDES
+        "${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/include"
+        "${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/config"
+        "${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/nrf_cc310_plat/include"
+    )
+endif()
+
+set(OT_PLATFORM_DEFINES ${OT_PLATFORM_DEFINES} PARENT_SCOPE)
+list(APPEND OT_PRIVATE_DEFINES "MBEDTLS_USER_CONFIG_FILE=\"nrf52840-mbedtls-config.h\"")
+list(APPEND OT_PUBLIC_INCLUDES
+    "${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/mbedtls_plat_config"
+)
+set(OT_PUBLIC_INCLUDES ${OT_PUBLIC_INCLUDES} PARENT_SCOPE)
+string(REPLACE "-pedantic-errors" "" OT_CFLAGS "${OT_CFLAGS}")
```

Comment:
We should not modify OT_CFLAGS here.
If we do need some extra CFLAGS to be set or unset, please create a new variable.